DigitalLearningSolutions.Data/Services/LoginService.cs

```diff
@@ -25,45 +25,39 @@ public LoginService(IUserService userService, IUserVerificationService userVerif
 
         public LoginResult AttemptLogin(string username, string password)
         {
-            var (unverifiedAdminUsers, unverifiedDelegateUsers) = userService.GetUsersByUsername(username);
+            var (unverifiedAdminUser, unverifiedDelegateUsers) = userService.GetUsersByUsername(username);
 
-            if (NoAccounts(unverifiedAdminUsers, unverifiedDelegateUsers))
+            if (NoAccounts(unverifiedAdminUser, unverifiedDelegateUsers))
             {
                 return new LoginResult(LoginAttemptResult.InvalidUsername);
             }
 
-            var (verifiedAdminUsers, verifiedDelegateUsers) = userVerificationService.VerifyUsers(
+            var (verifiedAdminUser, verifiedDelegateUsers) = userVerificationService.VerifyUsers(
                 password,
-                unverifiedAdminUsers,
+                unverifiedAdminUser,
                 unverifiedDelegateUsers
             );
 
-            var verifiedAdminUser = verifiedAdminUsers.SingleOrDefault();
-
             if (MultipleEmailsUsedAcrossAccounts(verifiedAdminUser, verifiedDelegateUsers))
             {
                 throw new LoginWithMultipleEmailsException("Not all accounts have the same email");
             }
 
-            var adminAccountVerificationAttemptedAndFailed = unverifiedAdminUsers.Any() && verifiedAdminUser == null;
-            var adminAccountToRecordFailureAgainstIsKnown = unverifiedAdminUsers.Count == 1;
+            var adminAccountVerificationAttemptedAndFailed = unverifiedAdminUser != null && verifiedAdminUser == null;
 
-            var adminAccountIsAlreadyLocked = adminAccountToRecordFailureAgainstIsKnown &&
-                                              unverifiedAdminUsers.Single().IsLocked;
-            var adminAccountHasJustBecomeLocked = adminAccountToRecordFailureAgainstIsKnown &&
-                                                  unverifiedAdminUsers.Single().FailedLoginCount == 4 &&
+            var adminAccountIsAlreadyLocked = unverifiedAdminUser?.IsLocked == true;
+            var adminAccountHasJustBecomeLocked = unverifiedAdminUser?.FailedLoginCount == 4 &&

```

Comment:
This is still an issue - `adminAccountHasJustBecomeLocked` can be true when `shouldIncreaseFailedLoginCount` is false (when we have verified delegates), so the code is lying.